src/main/scala/com/thoughtworks/deeplearning/sbtammoniteclasspath/AmmoniteClasspath.scala

```diff
@@ -16,37 +16,56 @@ object AmmoniteClasspath extends AutoPlugin {
   object autoImport {
     val exportToAmmoniteScript = taskKey[File](
       "Export classpath as a .sc file, which can be loaded by another ammonite script or a Jupyter Scala notebook")
+    val launchAmmoniteRepl = taskKey[Unit](
+      "Launch Ammonite REPL with classpath"
+    )
   }
   import autoImport._
 
-  override def projectSettings: Seq[Def.Setting[_]] = Seq {
+  override def projectSettings: Seq[Def.Setting[_]] = {
     val configuration = !Each(Seq(Compile, Test, Runtime))
     val classpathKey = !Each(Seq(fullClasspath, dependencyClasspath, managedClasspath, unmanagedClasspath))
 
-    exportToAmmoniteScript in classpathKey in configuration := {
-      val code = {
-        def ammonitePaths = List {
-          q"_root_.ammonite.ops.Path(${(!Each((classpathKey in configuration).value)).data.toString})"
-        }
+    Seq(
+      exportToAmmoniteScript in classpathKey in configuration := {
+        val code = {
+          def ammonitePaths = List {
+            q"_root_.ammonite.ops.Path(${(!Each((classpathKey in configuration).value)).data.toString})"
+          }
 
-        def mkdirs = List {
-          val ammonitePath = !Each(ammonitePaths)
-          q"""
-          if (!_root_.ammonite.ops.exists($ammonitePath)) {
-            _root_.ammonite.ops.mkdir($ammonitePath)
+          def mkdirs = List {
+            val ammonitePath = !Each(ammonitePaths)
+            q"""
+            if (!_root_.ammonite.ops.exists($ammonitePath)) {
+              _root_.ammonite.ops.mkdir($ammonitePath)
+            }
+            """
           }
+
+          q"""
+          ..$mkdirs
+          interp.load.cp(Seq(..$ammonitePaths))
           """
         }
+        val file = (crossTarget in configuration).value / s"${classpathKey.key.label}-${configuration.id}.sc"
+        IO.write(file, code.syntax)
+        file
+      },
+      launchAmmoniteRepl in classpathKey in configuration := {
+        val file = (exportToAmmoniteScript in classpathKey in configuration).value
+        import java.lang.{Process, ProcessBuilder}
 
-        q"""
-        ..$mkdirs
-        interp.load.cp(Seq(..$ammonitePaths))
-        """
+        System.out.synchronized {
+          println("Launching Ammonite REPL...");
+          new ProcessBuilder("amm", "--predef", file.toString)
```

Comment:
Would you mind implementing this feature in a more sophisticated way? Maybe something similar to this: https://github.com/alexarchambault/sbt-ammonite/blob/961c7c92626cb7697386708b3c553637478fdf0c/src/main/scala/ammonite/AmmonitePlugin.scala#L9

Reply:
Hmm, sure, I went with simplest solution I could think of but you are write about the separate process/installation argument. I'll try to incorporate sbt-ammonite.

Reply:
I have worked on this a little, I think I managed to get ammonite work with full classpath. But the code doesn't look any different from sbt-ammonite and also I couldn't get it to work with different classpaths (`dependencyClasspath`, `managedClasspath`, `unmanagedClasspath`) as in this repository; despite my efforts it always utilizes the full classpath. If you would like, I can share the code.

Reply:
You may find the code here:

https://github.com/ciuncan/sbt-ammonite-classpath/commit/5b5dacc5e86126f5128ec04b0df78efc9d7e4a21

It supports `Ammonite/run`, `Ammonite/fullClasspath/run`, ... `Ammonite/unmanagedClasspath`, but only first two works as intended. I read the sbt documentations and came to the conclusion that I will have to add new configurations (as in sbt-assembly) in order not to pollute user repos' `libraryDependencies`.

If you see any hope and can provide directions I can look into it. Otherwise I'll leave this here.

Reply:
Would it be better if we could support `Compile/fullClasspath/amm`, `Test/unmanagedClasspath/amm`, etc?

Reply:
I think the only safe way to pass classpath is `--predef-code` or `--predef`.

Reply:
I think we can generate code in `Ammonite/fullClasspath/initialCommands`, then load it with `--predef-code`

Reply:
Thank you for the directions. I have found two problems in my code:
- a silly error, I didn't iterate over the classpath list to generate classpath variants of `run` task, so it was always falling back to `Ammonite/run` with fullClasspath even though you try to invoke `Ammonite/unmanagedClasspath/run` task. :sweat_smile:) 
- the main reason why other classpath options didn't work: I have indeed added the ammonite dependency under Ammonite config, but if I, for example, use unmanagedClasspath I don't get Ammonite, because it is a managed dependency :). So, in run, I now, I discarded all the inherited libraryDependencies and set it to ammonite only, then always provide `Ammonite / managedClasspath` just to get ammonite repl. Then, as you suggested, I loaded the desired classpath with `--predef` command line arg while invoking main `ammonite.Main`. I have reused the existing export tasks, instead of using `--predef-code` with generated code. Is that OK?

As for `initialCommands`, wouldn't it be desirable to leave it to user repository's use? Maybe they would like to customize initial commands per classpath. I set `proj/Ammonite/initialCommands` to be the same as `proj/Compile/console/initialCommands`, but if this doesn't work for them, they will be able to override it with, for example, `proj/Ammonite/unmanagedClasspath/initialCommands` (because, the `console / initialCommands` might be referring to something in the repo source code which won't be available with `unmanagedClasspath`).

I think functionality-wise we've got the desired effect. We can look into tidying things, like code organization, the way of invoking the commands, settings, etc. I'll commit&push the code in couple of hours.

Reply:
Here is the latest commit, I tidied the code up a little, also tested with all combinations. They appear to work OK. https://github.com/ciuncan/sbt-ammonite-classpath/commit/b39805b250fced5d00582ba62419358babda2921

This one contains a script for convenience, runs `publishLocal` then replaces the line which contains this plugin in global sbt config with published version. https://github.com/ciuncan/sbt-ammonite-classpath/commit/2dbe2535bbbfafdd0b132b69ac5364cebf4a3cca

Once reviews are done, I'll squash all commits into one. I can leave the last one out if you want.

Reply:
Usage examples:
- `ammonite:run`
- `ammonite:unmanagedClasspath::run`
- `ammonite-test:run`
- `ammonite-test:unmanagedClasspath::run`
- `ammonite-runtime:run`
- ...